### PR TITLE
feat(data-warehouse): implement signoz import source

### PR DESCRIPTION
**Pull request by Gilbert09**
## Problem

SigNoz is an OpenTelemetry-native open-source observability platform (APM, logs, traces, metrics) with a hosted cloud offering. The `signoz` warehouse source existed only as a scaffolded stub, so users couldn't sync their SigNoz data into the PostHog data warehouse.

## Changes

Implements the SigNoz import source end to end, following the standard `source.py` / `settings.py` / `signoz.py` split:

- **Telemetry tables (`logs`, `traces`)** sync incrementally through `POST /api/v5/query_range` with `requestType: "raw"`. The epoch-millisecond `start`/`end` window params are a genuine server-side time filter, so incremental syncs advance from the stored `timestamp` watermark (30-day lookback on first sync, bounded by the account's retention). Rows are ordered ascending by `timestamp` plus a stable tiebreaker (`id` for logs, `span_id` for traces) per the SigNoz API docs.
- **Config tables (`alert_rules`, `dashboards`, `notification_channels`)** are small list endpoints with no server-side time filter, so they ship as full refresh.
- **Pagination**: query_range offset paging scans `offset + limit` rows server-side, so instead of an ever-growing offset the paginator advances the window start to the last row's timestamp after each full page and keeps only a small trailing-row offset. Rows re-listed at a window boundary are deduped by primary-key merge.
- **Resumable**: built on `ResumableSource`; resume state is the current window plus offset, saved after each yielded batch.
- **Auth and safety**: `SIGNOZ-API-KEY` header against the customer's tenant host. The host field is named `host`, so credential-retarget protection applies; hosts resolving to internal IPs are blocked at create and at sync time, and redirects are refused (SSRF). All HTTP goes through `make_tracked_session()` with the key redacted.
- 401/403 map to non-retryable errors with actionable messages; 429/5xx retry with `Retry-After`-aware backoff via tenacity.
- Adds `canonical_descriptions.py` (from the official API docs and the SigNoz source), the official SigNoz SVG icon, and moves `signoz` into the Implemented table in `SOURCES.md`.
- Metric series are intentionally out of scope: the raw request type only supports logs and traces, and metrics need a per-metric aggregation query that doesn't map to a generic warehouse table.

The source stays behind `unreleasedSource=True` with `releaseStatus=alpha` for now.

> [!NOTE]
> I couldn't curl-verify against a live SigNoz tenant (no credentials available), so endpoint behavior was verified against the public API docs and the SigNoz open-source query-service code (request/response types, default selected columns, pagination semantics). Conservative choices are marked with code comments where behavior couldn't be observed live.

The user-facing doc is in [PostHog/posthog.com#18465](https://github.com/PostHog/posthog.com/pull/18465) and `docsUrl` points at `/docs/cdp/sources/signoz`.

## How did you test this code?

Automated tests only (no live SigNoz tenant was available):

- `tests/test_signoz.py` (transport): window-advance paginator behavior including the same-millisecond page-boundary case (guards against skipped or duplicated rows), resume-from-saved-window, state saved only after yield (a crash re-yields rather than skips), v5 response envelope parsing, credential status mapping including 403-at-create vs 403-scoped, redirect and internal-host rejection, and epoch conversion for Go's nanosecond RFC 3339 timestamps.
- `tests/test_signoz_source.py` (source class): schema catalog and incremental flags, credential/argument plumbing, resumable manager binding.
- Ran the source-registry guard suites (`test_source_categories`, `test_source_config_generator`) and `ruff check`/`format`; `hogli ci:preflight --fix` passes.

## Automatic notifications

- [ ] Publish to changelog?
- [ ] Alert Sales and Marketing teams?

## Docs update

Doc added in the posthog.com repo (see Changes). `audit_source_docs` passes for signoz against that checkout.

## 🤖 Agent context

**Autonomy:** Human-driven (agent-assisted)

Authored with Claude Code (PostHog Code cloud task). Skills invoked: `/implementing-warehouse-sources`, `/writing-tests`, `/documenting-warehouse-sources`. Key decisions: modeled on the Datadog source (same observability shape) with Okta's arbitrary-host safety pattern; chose window-advance over plain offset paging after reading the SigNoz querier code (offset paging is quadratic server-side); skipped the v5 `cursor` param because its millisecond-granularity cursor can skip same-millisecond rows at page boundaries; excluded metrics (no raw row support). Icon is the official mark from the SigNoz repo (Apache-2.0).

---
*Created with [PostHog Code](https://posthog.com/code?ref=pr)*

### CODE REVIEW

File: products/warehouse_sources/backend/temporal/data_imports/sources/signoz/signoz.py [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+    # Re-check at run time (not just at source-create) in case the host was edited or now
+    # resolves to an internal address (SSRF / DNS rebinding). Only enforced on cloud.
+    host_ok, host_err = _is_host_safe(normalize_host(host), team_id)
+    if not host_ok:
+        raise SigNozHostNotAllowedError(host_err or HOST_NOT_ALLOWED_ERROR)
+
+    base = _base_url(host)
+    # One tracked session reused across pages and retries; the API key is redacted from
+    # logged URLs and captured samples.
+    session = make_tracked_session(headers=_get_headers(api_key), redact_values=(api_key,))
```
[the remark this answers is not in the document]
Comment (Gilbert09) [reply products/warehouse_sources/backend/temporal/data_imports/sources/signoz/signoz.py]: Fixed in 174dafa. Both the credential probe (`validate_credentials`) and the sync session (`get_rows`) now create their `make_tracked_session` with `capture=False`, so response bodies never enter HTTP sample capture. `redact_values=(api_key,)` and the no-redirect (SSRF) protections are unchanged.

Added regression tests asserting the session factory receives `capture=False` for both the validation probe and the sync path.

<sub>🦉 via [talyn.dev](https://talyn.dev)</sub>

File: products/warehouse_sources/backend/temporal/data_imports/sources/signoz/settings.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        name="dashboards",
+        kind="config",
+        path="/api/v1/dashboards",
+        primary_keys=("id",),
+        partition_key="createdAt",
+    ),
+    "notification_channels": SigNozEndpointConfig(
+        name="notification_channels",
+        kind="config",
+        path="/api/v1/channels",
```
[the remark this answers is not in the document]
Comment (Gilbert09) [reply products/warehouse_sources/backend/temporal/data_imports/sources/signoz/settings.py:102]: Fixed in 174dafa. `notification_channels` records now pass through a strict allowlist (`allowed_fields=("id", "name", "type", "createdAt", "updatedAt")`) applied in `_extract_config_items`, so the credential-bearing `data` payload (Slack webhook URLs, PagerDuty keys, etc.) is dropped before any record is yielded and never lands in the warehouse table.

I kept the safe metadata (which channels exist and their types) rather than dropping the endpoint entirely, which matches the allowlist option you suggested. Also removed the `data` column from the canonical descriptions and added regression tests at both the extractor level and the end-to-end sync path.

<sub>🦉 via [talyn.dev](https://talyn.dev)</sub>

**Comment by Gilbert09** [comment]
### CI note: `deploy preview` failure is infra flakiness, not the diff

The only red check is `deploy preview`, which fails in its **Bring up preview** step while provisioning the ephemeral preview environment — nothing in this PR's code path is exercised there. I re-ran it once and it failed a second time with a *different* infra error, which confirms it's non-deterministic environment flakiness rather than a real regression:

- First run: `hogland API error (HTTP 500): placement failed … chunkfs mount: chunkfsd ready: not ready within 5s`
- Re-run: `httpx.ReadTimeout: The read operation timed out` (talking to the same provisioning API)

`deploy preview` is not in the branch's required status checks, and this PR is backend-only (Python warehouse-source files) with no frontend/deploy surface. All required checks — backend, Django, dagster, visual regression, lint, etc. — are green. Leaving this as-is since the root cause is the preview-provisioning service, which isn't fixable from this branch.

<sub>🦉 via [talyn.dev](https://talyn.dev)</sub>

**Review by danielcarletti — APPROVED** [review]
LGTM
